### ISSUE TITLE
Ambiguity of format for template fields in Airflow Kubernetes Operator (airflow.providers.cncf.kubernetes.operators.kubernetes_pod)

### ISSUE THREAD

**Issue by ybressler** [apache/airflow#22154]
### Apache Airflow Provider(s)

amazon

### Versions of Apache Airflow Providers

3.0.2 (https://airflow.apache.org/docs/apache-airflow-providers-cncf-kubernetes/stable/_modules/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.html)

### Apache Airflow version

2.2.4 (latest released)

### Operating System

NA

### Deployment

MWAA

### Deployment details

NA

### What happened

Documentation for [`airflow.providers.cncf.kubernetes.operators.kubernetes_pod`](https://airflow.apache.org/docs/apache-airflow-providers-cncf-kubernetes/stable/_api/airflow/providers/cncf/kubernetes/operators/kubernetes_pod/index.html#airflow.providers.cncf.kubernetes.operators.kubernetes_pod.KubernetesPodOperator) suggests that [`template_fields`](https://airflow.apache.org/docs/apache-airflow-providers-cncf-kubernetes/stable/_api/airflow/providers/cncf/kubernetes/operators/kubernetes_pod/index.html#airflow.providers.cncf.kubernetes.operators.kubernetes_pod.KubernetesPodOperator.template_fields) are a list of string values. This is accordance with Airflow's custom operator guides, to create a class attribute with a list of string values. However, when consulting the [source code](https://airflow.apache.org/docs/apache-airflow-providers-cncf-kubernetes/stable/_modules/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.html#KubernetesPodOperator.template_fields), the values for `template_fields` are stored in a tuple:
```
template_fields: Sequence[str] = (
        'image',
        'cmds',
        'arguments',
        'env_vars',
        'labels',
        'config_file',
        'pod_template_file',
        'namespace',

    )
```
This is problematic from a documentation standpoint, since it is unclear how to extend behaviors with ambiguous typing. For example, I will encounter a runtime error when extending template fields as such:
```
from airflow.providers.cncf.kubernetes.operators.kubernetes_pod import KubernetesPodOperator

class CustomKubernetesPodOperator(KubernetesPodOperator):
    # The following will fail
    template_fields = KubernetesPodOperator.template_fields + ["name"]

    def __init__(self):
        super.__init__()
```
My solution is to wrap the template_fields in a `list` constructor as such:
```
template_fields = list(KubernetesPodOperator.template_fields) + ["name"]
```

But this is not an elegant solution. Plus, documentation was not helpful. A solution would be to either clarify the documentation (that this is stored as a tuple default), or convert to storing as a list. (As typical with other Airflow operators.)

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by uranusjr** [comment]
We should rewrite the documentation to say the attribute is only guaranteed to be a _sequence_, and maybe add some documentation to recommend using `*` instead. For your example, this is the safe syntax (without the overhead of creating a temporary list):

```python
template_fields = [*KubernetesPodOperator.template_fields, "name"]
```

**Comment by uranusjr** [comment]
Hmm, this is weird. The documentation is automatically generated from AutoAPI, and I have no idea why it chooses to render the tuple as a list. Perhaps we need to raise this issue to AutoAPI. https://github.com/readthedocs/sphinx-autoapi

**Comment by uranusjr** [comment]
Done in readthedocs/sphinx-autoapi#330

**Comment by ybressler** [comment]
Good solution. And yes, would be even more helpful for describing how to append to template fields with this unpacking approach. Thanks for being on top of this @uranusjr

**Comment by josh-fell** [comment]
The [fix for the issue @uranusjr opened](https://github.com/readthedocs/sphinx-autoapi/commit/847b6e5626dc3e85ab7e365c1893a8525a2af90a) was finally released on 2023-01-17 with [v2.0.1](https://pypi.org/project/sphinx-autoapi/2.0.1/). The proper rendering of `template_fields` should be fixed when the provider's API documentation is generated again.

I can confirm that building the docs locally with `sphinx-autoapi==2.0.1` does render the `template_fields` value as expected.
<img width="922" alt="image" src="https://user-images.githubusercontent.com/48934154/214215125-454916a5-63df-4e57-9bf5-9b07d2f07834.png">

Perhaps we should close this and open a new issue for some docs enhancements re: template fields based on this comments:
>We should rewrite the documentation to say the attribute is only guaranteed to be a sequence, and maybe add some documentation to recommend using * instead. For your example, this is the safe syntax (without the overhead of creating a temporary list)...

WDYT @uranusjr?

**Comment by uranusjr** [comment]
Sound good to me.

**Comment by eladkal** [comment]
Docs were regenerated as we just released kubernetes provider 5.2.0

**before:**

![Screenshot 2023-02-11 at 23 00 59](https://user-images.githubusercontent.com/45845474/218281049-0a01f1a9-2ca3-47ec-836a-39116e76ca98.png)


**after:**

![Screenshot 2023-02-11 at 23 01 09](https://user-images.githubusercontent.com/45845474/218281054-297b4874-4b99-45bc-9e4e-150010048b7c.png)


@josh-fell would you like to take the followup doc task?

**Comment by josh-fell** [comment]
> Docs were regenerated as we just released kubernetes provider 5.2.0
> 
> **before:**
> 
> ![Screenshot 2023-02-11 at 23 00 59](https://user-images.githubusercontent.com/45845474/218281049-0a01f1a9-2ca3-47ec-836a-39116e76ca98.png)
> 
> **after:**
> 
> ![Screenshot 2023-02-11 at 23 01 09](https://user-images.githubusercontent.com/45845474/218281054-297b4874-4b99-45bc-9e4e-150010048b7c.png)
> 
> @josh-fell would you like to take the followup doc task?

Yes sir, sign me up. It's on my list.

**Comment by josh-fell** [comment]
As of version 5.2.0, the formatting of `template_fields` in the Python API docs has been corrected via https://github.com/readthedocs/sphinx-autoapi/issues/330. Further documentation for `template_fields` generally will be updated per the [above comment](https://github.com/apache/airflow/issues/22154#issuecomment-1065394841). Closing.
